warehouse/accounts/views.py

```diff
@@ -252,9 +252,12 @@ def request_password_reset(request, _form_class=RequestPasswordResetForm):
     form = _form_class(request.POST, user_service=user_service)
 
     if request.method == "POST" and form.validate():
-        user = user_service.get_user_by_username(form.username.data)
-        fields = send_password_reset_email(request, user)
-        return {'n_hours': fields['n_hours']}
+        user = user_service.get_user_by_username(form.username_or_email.data)
+        if user is None:
+            user = user_service.get_user_by_email(form.username_or_email.data)
+        if user is not None:
```

Comment:
This can just be `if user:`

Reply:
or it can be just `else:` :)

Reply:
@pradyunsg Actually, we do want to test `user` again, since `get_user_by_email` could return either valid user or `None`, and we don't want to send the email if it's `None`.

Reply:
Whoops! I missed the assignment in the body.

_hides behind a paper bag_